designs/CentrallyManagingNuGetPackageVersions-VSManagePackages.md

```diff
@@ -0,0 +1,414 @@
+4. When uninstalling/removing a package from a centrally managed project, the `PackageReference` is removed from the project file, but the `PackageVersion` is kept in the `Directory.Packages.props` file.
+
+## Later Phase Goals
+
+5. Removing a `PackageVersion` from the `Directory.Packages.props` file is out of scope for the first version, but will be a requirement in the future.
+
+## Non - Goals
+
+6. This spec is scoped to Visual Studio API interaction between components. Package Manager UI changes are out of scope. Command Line Interface is out of scope.
+7. `Directory.Packages.props` files that import other MSBuild files. If the customer uses this, the behavior is undefined, specifically around which `Directory.Packages.props` file gets updated.
+
+## Scenarios
+
+The following scenarios will be enabled when the proposed design changes will be completed.
+
+### Install a package in a project that already exists in `Directory.Packages.props`
+
+In this scenario, there is a `Directory.Packages.props` file which specifies a `PackageVersion` for `PackageA`. There is a project `Project1`, which does not currently have a reference to `PackageA`. When the customer installs `PackageA` into `Project1`, the project's file is modified to include a `<PackageReference Include="PackageA" />`, which does not specify a version. `Directory.Packages.props` remains unchanged.
+
+#### Before
+
+`$(repo_root)\Directory.Packages.props`
+
+```xml
+<Project>
+  <ItemGroup>
+    <PackageVersion Include="PackageA" Version="1.2.3" />
+  </ItemGroup>
+</Project>
+```
+
+`$(repo_root)\Project1\Project1.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+</Project>
+```
+
+#### Gesture
+
+Customer installs `PackageA` into `Project1` using Package Manager UI.
+
+#### After
+
+`$(repo_root)\Directory.Packages.props`
+
+```xml
+<Project>
+  <ItemGroup>
+    <PackageVersion Include="PackageA" Version="1.2.3" />
+  </ItemGroup>
+</Project>
+```
+
+`$(repo_root)\Project1\Project1.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageA" />
+  </ItemGroup>
+
+</Project>
+```
+
+### Install a package in a project that does not exist in `Directory.Packages.props`
+
+In this scenario, there is a `Directory.Packages.props` file, but it does not have `PackageVersion` for `PackageA`. There is a project `Project1`, which also does not currently have a reference to `PackageA`. When the customer installs `PackageA`, version `1.2.3`, into `Project1`, `Directory.Packages.props` is modified to include `<PackageVersion Include="PackageA" Version="1.2.3" />`, and the project's file is modified to include a `<PackageReference Include="PackageA" />`.
```

Comment:
It will be the closest Directory.Packages.props in the folder structure that will be modified.